halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Utils.java

```diff
@@ -51,13 +46,14 @@
 import java.util.Map;
 
 @Slf4j
+@Component
```

Comment:
This seemed like the cleanest workaround to the static autowire issue explained above. Another kind of clunky alternative is to leave `secretSessionManager` as static and add a class that creates an autowired `secretSessionManager` and sets the static one in `KubernetesV2Utils` like this:

```
@Component
public class StaticSecretSessionManagerInitializer {

  @Autowired
  private SecretSessionManager secretSessionManager;

  @PostConstruct
  public void init() {
    KubernetesV2Utils.setStaticSecretSessionManager(secretSessionManager);
  }
}
```

Reply:
As for the specifics in the changes: So far all secrets are being decrypted by autowiring a `SecretSessionManager` which contains the decrypt methods. This works well in all cases but kubeconfig decryption. The problematic spot is in `KubernetesV2Utils`, since the methods are all static and since we can't call the autowired `SecretSessionManager` from a static context, we settled on making the methods non-static.

In most of the places `KubernetesV2Utils` is used, we could just replace the static calls with an autowired `kubernetesV2Utils`. The one where this didn't work (and where it probably gets confusing) is in `KubernetesV2Service` since it's an interface. So there, we use the autowired `SecretSessionManager` that's already hooked up in `KubernetesV2Executor`.

Reply:
Agree that having `Utils` in the name is not ideal but that predates the secrets work and fixing the naming should be out of scope for this PR.